Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -78,45 +92,49 @@ void preparePathForEditing() {
             // Open a new path that the contents of the old path will be streamed into. We cannot re-use the same path as it is now closed
             // and cannot be opened again. We use the newPath variable because the factory was returning the same pointer for some strange
             // reason so this will force it to do otherwise.
-            FAIL_FAST_IF_FAILED(factory->CreatePathGeometry(&newPath));
-            FAIL_FAST_IF_FAILED(newPath->Open(&newSink));
-            FAIL_FAST_IF_FAILED(_impl.pathGeometry->Stream(newSink.Get()));
+            RETURN_IF_FAILED(factory->CreatePathGeometry(&newPath));
+            RETURN_IF_FAILED(newPath->Open(&newSink));
+            RETURN_IF_FAILED(PathGeometry()->Stream(newSink.Get()));
 
-            _impl.pathGeometry = newPath;
-            _impl.geometrySink = newSink;
+            PathGeometry() = newPath;
+            GeometrySink() = newSink;
```

Comment:
This generally seems unconventional. Although it makes sense in code terms. It might be better to have a setter?
Also Why don't you use move this code into the actual implementation and make this a forwarding call, that way you provide better encapsulation and have the majority of the logic attached with the implementation.